### ISSUE TITLE
Async training gets stuck in single-agent cases.

### ISSUE THREAD

**Issue by KornbergFresnel** [sjtu-marl/malib#8]
**branch**: vector-env

See the attached, there is only 1 training epoch for 100 rollout epochs.

**script**: `python examples/async_simple.py`

![image](https://user-images.githubusercontent.com/18256149/124719419-b0f66600-df39-11eb-866c-27f2b9b1763a.png)

**Comment by VegeWong** [comment]
Have fixed it in the latest commit on vector-env branch(https://github.com/sjtu-marl/malib/commit/b65e5c09407e18d9a5b01c35d87de18f3839197c). The problem is raised from that updating parameters via aggregation of gradients does not respect settings of learning rates. Since it is a minor fix, the patch is directly pushed to the branch. It will be appreciated if you can recheck the problem.
![25511626959222_ pic_hd](https://user-images.githubusercontent.com/24775547/126649878-fc21e065-bb2e-4a66-bef0-790278c878c5.jpg)

**Comment by zbzhu99** [comment]
> Have fixed it in the latest commit on vector-env branch([b65e5c0](https://github.com/sjtu-marl/malib/commit/b65e5c09407e18d9a5b01c35d87de18f3839197c)). The problem is raised from that updating parameters via aggregation of gradients does not respect settings of learning rates. Since it is a minor fix, the patch is directly pushed to the branch. It will be appreciated if you can recheck the problem.
> ![25511626959222_ pic_hd](https://user-images.githubusercontent.com/24775547/126649878-fc21e065-bb2e-4a66-bef0-790278c878c5.jpg)

I'm curious as to why this can block the training process? It seems to me that the change made in the commit only affects the magnitude of logged gradients.

**Comment by KornbergFresnel** [comment]
Fixed it in PR #14. The blocking is caused by unbalanced read/write access to `OfflineDatasetServer`, which has been alleviated in #5 by using [Readers-writer lock](https://en.wikipedia.org/wiki/Readers%E2%80%93writer_lock).

**Comment by VegeWong** [comment]
> > Have fixed it in the latest commit on vector-env branch([b65e5c0](https://github.com/sjtu-marl/malib/commit/b65e5c09407e18d9a5b01c35d87de18f3839197c)). The problem is raised from that updating parameters via aggregation of gradients does not respect settings of learning rates. Since it is a minor fix, the patch is directly pushed to the branch. It will be appreciated if you can recheck the problem.
> > ![25511626959222_ pic_hd](https://user-images.githubusercontent.com/24775547/126649878-fc21e065-bb2e-4a66-bef0-790278c878c5.jpg)
> 
> I'm curious as to why this can block the training process? It seems to me that the change made in the commit only affects the magnitude of logged gradients.

As is shown from my test cases, the false gradients generating logic caused the gradients explosion after several rounds of optimization, resulting in abnormal value like "none". These kinds of values ("nan" and "inf") are blocked by the TensorboardX backend, stalling the status update of involved graphs in the frontend webpage.
